### DISCUSSION TITLE
SSH agent not working under niri

### DISCUSSION THREAD

**Discussion by cmvanb** [YaLTeR/niri#769] [Q&A]
With another wayland compositor (river) running ssh-agent.service has been sufficient. However, no shell spawned by niri can remember my password.

The service is running, and SSH_AUTH_SOCK is in my shell environment.

`systemctl --user status ssh-agent.service`
```
● ssh-agent.service - OpenSSH key agent
     Loaded: loaded (/usr/lib/systemd/user/ssh-agent.service; enabled; preset: enabled)
     Active: active (running) since Thu 2024-10-31 09:33:57 CET; 12h ago
 Invocation: abdcf89869514b829e2a0892a139c1a7
       Docs: man:ssh-agent(1)
             man:ssh-add(1)
             man:ssh(1)
   Main PID: 1485 (ssh-agent)
      Tasks: 1 (limit: 76720)
     Memory: 1.3M (peak: 2M)
        CPU: 12ms
     CGroup: /user.slice/user-1000.slice/user@1000.service/app.slice/ssh-agent.service
             └─1485 /usr/bin/ssh-agent -D -a /run/user/1000/ssh-agent.socket

Oct 31 09:33:57 cyxwel systemd[1470]: Started OpenSSH key agent.
Oct 31 09:33:57 cyxwel ssh-agent[1485]: SSH_AUTH_SOCK=/run/user/1000/ssh-agent.socket; export SSH_AUTH_SOCK;
Oct 31 09:33:57 cyxwel ssh-agent[1485]: echo Agent pid 1485;
```

Am I supposed to configure something for niri to see the agent?

**Comment by YaLTeR** [comment]
Not sure. My ssh agent seems to work fine. Check that the environment variable is indeed accessible in the terminals that you're using. 

niri-session goes through a login shell so it should get all environment variables from shell configuration.

**Comment by cmvanb** [comment]
The environment variable is definitely available in the terminals spawned by niri.

Do you use Gnome Keyring? or the OpenSSH agent?

**Reply by YaLTeR** [reply]
I use GNOME keyring. I have gcr-ssh-agent.socket enabled.

**Comment by BeringLogic** [comment]
I had the same problem and got it working by adding this to my config.kdl:

```
environment {
    SSH_AUTH_SOCK "/run/user/1000/ssh-agent.socket"
}
```

Is there a way to use another variable? The proper way would be this instead but it's not working:
```
environment {
    SSH_AUTH_SOCK "${XDG_RUNTIME_DIR}/ssh-agent.socket"
}
```

**Reply by YaLTeR** [reply]
You can put this in your login shell config, i.e. ~/.bash_profile
